PureInvoke/Functions/Invoke-AdvapiLookupAccountSid.ps1

```diff
@@ -0,0 +1,69 @@
+
+function Invoke-AdvApiLookupAccountSid
+{
+    <#
+    .SYNOPSIS
+    Calls the Advanced Windows 32 Base API (advapi32.dll) `LookupAccountSid` function.
+
+    .DESCRIPTION
+    The `Invoke-AdvApiLookupAccountSid` function calls the advapi32.dll API's `LookupAccountSid` function, which looks up a
+    SID and returns its account name, domain name, and use. Pass the SID as a byte array to the `Sid` parameter and the
+    system name to the `SystemName` parameter, which are passed to `LookupAccountSid` as the `Sid` and `lpSystemName`
+    arguments, respectively. The function returns an object with properties for each of the `LookupAccountSid`
+    function's out parameters: `Name`, `ReferencedDomainName`, and `Use`.
+
+    .LINK
+    https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-lookupaccountsida
+
+    .EXAMPLE
+    Invoke-AdvApiLookupAccountSid -Sid $sid
+
+    Demonstrates how to call this function by passing a sid to the `Sid` parameter.
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        [byte[]] $Sid,
+
+        [String] $SystemName
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState
+
+    [StringBuilder] $name = [StringBuilder]::New()
+    # cch = count of chars
+    [UInt32] $cchName = $name.Capacity;
+
+    [StringBuilder] $domainName = [StringBuilder]::New()
+    [UInt32] $cchDomainName = $domainName.Capacity;
+
+    [SidNameUse] $sidNameUse = [SidNameUse]::Unknown;
+
+    $result = [AdvApi32]::LookupAccountSid($SystemName, $sid, $name, [ref] $cchName, $domainName, [ref] $cchDomainName,
+                                           [ref] $sidNameUse)
+    $errCode = [Marshal]::GetLastWin32Error()

```

Comment:
Should this statement be nested within the `(-not $result)` condition?

Reply:
Ditto. Best to capture the last error before something else sets/changes it.